src/esportsbot/cogs/MusicCog.py

```diff
@@ -0,0 +1,1117 @@
+            # Can't pause if the bot isn't playing
+            voice_client: VoiceClient = self._currently_active.get(guild_id).get('voice_client')
+            voice_client.pause()
+
+    def __resume_song(self, guild_id):
+        """
+        Resumes the playback of a specific guild if the guild is paused. Otherwise nothing.
+        :param guild_id: The id of the guild to resume the playback in.
+        """
+
+        if self._currently_active.get(guild_id).get('voice_client').is_paused():
+            # Only able to resume if currently paused
+            voice_client: VoiceClient = self._currently_active.get(guild_id).get('voice_client')
+            voice_client.resume()
+
+    def __check_valid_user_vc(self, ctx: Context):
+        """
+        Checks if the user: A) Is in a voice channel, B) The voice channel is the same as the voice channel the bot is
+        connected to, C) The message sent was in the music text channel.
+        :param ctx: The context of the message sent.
+        :return: If all the above conditions are met, True, otherwise False.
+        """
+
+        music_channel_in_db = db_gateway().get('music_channels', params={'guild_id': ctx.guild.id})
+        if ctx.message.channel.id != music_channel_in_db[0].get('channel_id'):
+            # Message is not in the songs channel
+            return False
+
+        if not ctx.author.voice:
+            # User is not in a voice channel
+            return False
+
+        if self._currently_active.get(ctx.guild.id).get('channel_id') != ctx.author.voice.channel.id:
+            # The user is not in the same voice channel as the bot
+            return False
+
+        return True
+
+    def __determine_message_type(self, message) -> int:
+        """
+        Determine if the message received is a video url, playlist url or a string that needs to be queried.
+        :param message: The message to determine the type of.
+        :return: An integer representing the message type.
+        """
+
+        if self.__is_url(message):
+            # The message is a url
+            if re.search(r'(playlist)', message):
+                # The message is a playlist
+                return 0
+            else:
+                return 1
+        else:
+            # The message is a string
+            return 2
+
+    def __get_opus_stream(self, formats):
+        """
+        Get the opus formatted streaming link from the formats dictionary.
+        :param formats: The formats dictionary that contains the different streaming links.
+        :return: A streaming url that links to an opus stream and the bit rate of the stream.
+        """
+
+        # Limit the codecs to just opus, as that is required for streaming audio
+        opus_formats = [x for x in formats if x.get('acodec') == 'opus']
+
+        # Sort the formats from highest br to lowest
+        sorted_opus = list(sorted(opus_formats, key=lambda k: float(k.get('abr')), reverse=True))
```

Comment:
instead of:
```
        sorted_opus = list(sorted(opus_formats, key=lambda k: float(k.get('abr')), reverse=True))
        chosen_stream = sorted_opus[self._bitrate_quality]
```

You could do:
```
        chosen_stream = max(opus_formats, key=lambda k: float(k.get('abr')))
```

Reply:
I've not changed this yet as it would eliminate the ability to choose which quality of audio stream we want.